dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestControllerIT.java

```diff
@@ -425,5 +514,127 @@ private int getNumberOfPdfPages(byte[] content) throws IOException {
+
+        resourcePolicy = resourcePolicyService.create(context);
+        resourcePolicy.setEPerson(eperson);
+        resourcePolicy.setAction(WRITE);
+        resourcePolicy.setdSpaceObject(bitstream);
+
+        bitstream = context.reloadEntity(bitstream);
+        bitstreamService.addMetadata(context, bitstream, "dc", "format", null, null, "test format");
+
+        context.commit();
+
+        getClient(getAuthToken(eperson.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
+                        )
+        ).andExpect(status().isOk());
+
+        assertThat(knownFormat, equalTo(context.reloadEntity(bitstream).getFormat(context)));
+        assertTrue(isEmpty(
+                bitstreamService.getMetadataByMetadataString(context.reloadEntity(bitstream), "dc.format")
+        ));
+
+        bitstream = context.reloadEntity(bitstream);
+        bitstreamService.addMetadata(context, bitstream, "dc", "format", null, null, "another test format");
+
+        context.commit();
+
+        getClient(getAuthToken(admin.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + unknownFormat.getID()
+                        )
+        ).andExpect(status().isOk());
+
+        getClient(getAuthToken(admin.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + randomUUID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + unknownFormat.getID()
+                        )
+        ).andExpect(status().isNotFound());
+
+        getClient(getAuthToken(admin.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/-1"
+                        )
+        ).andExpect(status().isBadRequest());
+
+        getClient(getAuthToken(admin.getEmail(), password)).perform(
+                put("/api/core/bitstreams/" + bitstream.getID() + "/format")
+                        .contentType(parseMediaType(TEXT_URI_LIST_VALUE))
+                        .content(
+                                "https://localhost:8080/spring-rest/api/core/bitstreamformat/" + knownFormat.getID()
```

Comment:
you should use `org.dspace.app.rest.test.AbstractControllerIntegrationTest.REST_SERVER_URL`